### ISSUE TITLE
-pthread not found on Mac OS X 11.6 (Big Sur), qs package does not build

### ISSUE THREAD

**Issue by PMSeitzer** [traversc/qs#63]
I was using another package that uses `qs::qread()`, and `qs::qread()` was crashing.  To debug the crash, I wanted to rebuild the `qs` package from source, and add some debugging print statements along the way to try to diagnose the problem.

However, when I cloned the `qs` repo and rebuilt the package using `R CMD INSTALL qs`, the package did not build, complaining that the linker could not find `-pthread`.

I'm running on mac os x 11.6 (Big Sur).  Apple made a bunch of changes to low-level system libraries that broke lots of things, and caused lots of problems. Fortunately, I was able to fix the problem by adding two lines to the `MakeVars.in` file:

```
# changes necessary to work with Big Sur
PKG_LIBS+= -F/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreFoundation.framework/..
PKG_LIBS+= -L/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib
```

Not sure @traversc if you wanted to address the issue by adjusting `MakeVars.in` or if you wanted to address the problem a different way.

BTW, once I was able to rebuild `qs` from source, the other package I was using that called `qs::qread()` started working properly again - so the bug doesn't appear to be related to `qs::qread()`, or at least, everything is working as expected in the latest version.

**Comment by traversc** [comment]
That's a big yikes. If pthreads isn't universally available, then the guidance in "Writing R Extensions" probably is outdated. 

For `qs`, I think the long term solution would be to switch to `RcppParallel` but that would require a re-factor. 

I plan on forwarding this issue to the mailing list and seeing what people think.

**Comment by PMSeitzer** [comment]
@traversc Great! glad you're working on the issue.

I'm happy to provide additional information if it's helpful.  Best of luck in forwarding the issue to the mailing list.

**Comment by traversc** [comment]
I upgraded Catalina -> Big Sur, but have not been able to reproduce the error. I commented out everything in ~/.R/Makevars just in case. 

Here's my clang:
```
% clang++ --version
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin20.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

Maybe something about xcode needs to be updated?

**Comment by PMSeitzer** [comment]
Hey @traversc , It's certainly possible that there is something strange about my build configuration that wouldn't affect users in general.   I am running on a different version of `clang++`, which might be the issue:
```
% clang++ --version
clang version 6.0.0 (tags/RELEASE_600/final)
Target: x86_64-apple-darwin20.6.0
Thread model: posix
InstalledDir: /usr/local/clang6/bin
```

**Comment by traversc** [comment]
Clang 6 is pretty old and I am far from an expert on Mac, but your clang isn't using xcode which is a little strange to me. 

The official R documentation recommends building xcode:
```
sudo xcode-select --install
```
https://mac.r-project.org/tools/

I wonder if that would help?

**Comment by PMSeitzer** [comment]
Hey @traversc ,
Thanks for the suggestion, but the I already have the xcode command line tools, and I've run the command you've suggested before.   I'm also using the `llvm` compiler for some programs, and I've had to mess with some low-level configuration files to manage multiple compilers.  It might be safe to say that this issue is probably specific to my system, and not a big problem.  I'm not sure why `qread()` was crashing, but I'm glad I could get it all to work once I rebuilt the package.  Thanks again for all of your help!
